src/abstractdataframe/abstractdataframe.jl

```diff
@@ -365,39 +369,58 @@ If the column does not allow missing values, `nothing` is returned.
 Consequently, `nmissing = 0` indicates that the column allows
 missing values, but does not currently contain any.
 
+Custom functions perform call `skipmissing` on columns of eltype `Union{T, Missing}`, 
+users should be aware that because of this, they cannot specify special handling
+of missing values in any of the functions given via a `Pair` argument.
+
+
 **Examples**
 
 ```julia
 df = DataFrame(i = 1:10, x = rand(10), y = rand(["a", "b", "c"], 10))
 describe(df)
 describe(df, stats = :all)
-describe(df, stats = [:min, :max])
+describe(df, :min, :max)
+describe(df, :mean, :variance => Statistics.var)
 ```
 
 """
-function StatsBase.describe(df::AbstractDataFrame; stats::Union{Symbol,AbstractVector{Symbol}} =
-                            [:mean, :min, :median, :max, :nunique, :nmissing, :eltype])
-    # Check that people don't specify the wrong fields.
+StatsBase.describe(df::AbstractDataFrame) = _describe(df, [:mean, :min, :median, 
+                                                     :max, :nunique, :nmissing, 
+                                                     :eltype])
+function StatsBase.describe(df::AbstractDataFrame, stats::Union{Symbol, Pair{Symbol}}...)
+    _describe(df, collect(stats))
+end
+
+function _describe(df::AbstractDataFrame, stats::Vector)   
+    predefined_funs = Symbol[s for s in stats if s isa Symbol] 
+
     allowed_fields = [:mean, :std, :min, :q25, :median, :q75,
                       :max, :nunique, :nmissing, :first, :last, :eltype]
-    if stats == :all
-        stats = allowed_fields
-    end
 
-    if stats isa Symbol
-        if !(stats in allowed_fields)
+    if predefined_funs == [:all]
+        predefined_funs = allowed_fields
+        i = findfirst(s -> s == :all, stats)
+        splice!(stats, i, allowed_fields) # insert in the stats vector to get a good order
+    elseif :all in predefined_funs 
+            throw(ArgumentError("If the user specifies `:all` it must be the only `Symbol` argument.")) 
```

Comment:
Avoid talking about "the user" to the user. :-) Anyway, avoid any unnecessary text.
```suggestion
            throw(ArgumentError("`:all` must be the only `Symbol` argument.")) 
```

Reply:
fixed.